pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,347 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plant
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	gardencorelisters "github.com/gardener/gardener/pkg/client/core/listers/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/utils"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	kubernetesclientset "k8s.io/client-go/kubernetes"
+	kubecorev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+)
+
+const kubeconfigChecksumAnnotationKey = "kubeconfig.secret.checksum/value"
+
+// plantAdd adds the plant resource
+func (c *Controller) updateClients(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+	c.plantQueue.Add(key)
+}
+
+// plantAdd adds the plant resource
+func (c *Controller) plantAdd(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+	c.plantQueue.Add(key)
+}
+
+// plantUpdate updates the plant resource
+func (c *Controller) plantUpdate(oldObj, newObj interface{}) {
+	oldPlant, ok1 := oldObj.(*gardencorev1alpha1.Plant)
+	newPlant, ok2 := newObj.(*gardencorev1alpha1.Plant)
+	if !ok1 || !ok2 {
+		return
+	}
+	oldPlantJSON, _ := json.Marshal(oldPlant)
+	newPlantJSON, _ := json.Marshal(newPlant)
+	plantLogger := logger.NewShootLogger(logger.Logger, newPlant.ObjectMeta.Name, newPlant.ObjectMeta.Namespace, "")
+
+	plantLogger.Debugf(string(oldPlantJSON))
+	plantLogger.Debugf(string(newPlantJSON))
```

Comment:
Let's remove that, I know that you copied from other controllers but it was never used so far. It's anyways inconsistent across all our controllers.